### PR TITLE
Fix memory leak in memcache

### PR DESCRIPTION
**Pull request by dangershony**

### CODE REVIEW

File: src/Stratis.Bitcoin/Utilities/MemoryCache.cs [hunk 1]
```diff
@@ -65,6 +65,16 @@ public MemoryCache(IEqualityComparer<TKey> comparer = null)
         /// <returns><c>true</c> if cache contains the item, <c>false</c> otherwise.</returns>
         protected abstract bool IsCacheFull(CacheItem item);
 
+        /// <summary>And item was added to the cache.</summary>
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/Utilities/MemoryCache.cs]: "An item was added..."

File: src/Stratis.Bitcoin/Utilities/MemoryCache.cs [hunk 2]
```diff
@@ -65,6 +65,16 @@ public MemoryCache(IEqualityComparer<TKey> comparer = null)
...
+        protected virtual void ItemAddedLocked(CacheItem item)
+        {
+        }
+
+        /// <summary>And item was removed from the cache.</summary>
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/Utilities/MemoryCache.cs]: "An item was added..."

File: src/Stratis.Bitcoin/Utilities/MemoryCache.cs (hunk 1 above)
Comment (dangershony) [reply src/Stratis.Bitcoin/Utilities/MemoryCache.cs]: ah well spotted

**Comment by dangershony** [comment]
Add also better log of cache size
image

**Pull request merged**